sqlmesh/dbt/common.py

```diff
@@ -19,6 +19,7 @@
 T = t.TypeVar("T", bound="GeneralConfig")
 
 PROJECT_FILENAME = DBT_PROJECT_FILENAME
+DBT_ALL_MODEL_ATTRS = "__DBT_ALL_MODEL_ATTRS__"
```

Comment:
I don't like the idea of bespoke placeholders. Can we use a union type or a dataclass instead? 
```
@dataclass
class AllModelAttrs:
    pass

model_attrs: t.Union[t.Set[str], AllModelAttrs]
```

or

```
@dataclass
class ModelAttrs:
    attrs: t.Set[str]
    all_attrs: bool = False

model_attrs: ModelAttrs
```

Reply:
Sure, it's a small refactor. Could you elaborate a bit on why you dislike it? I considered using a `Union` too but it felt like it would make the implementation a bit more involved without providing obvious advantage.

Reply:
Refactored: https://github.com/TobikoData/sqlmesh/pull/5269/commits/eea374718d4d343f7791dbdd2f80ab29fa3d3e41